### DISCUSSION TITLE
PDF file

### DISCUSSION THREAD

**Discussion by rafikApp** [Budibase/budibase#6476] [Help] [answered]
can we create a pdf file from a form ?

**Comment by melohagan** [comment, marked as answer]
Not as yet, but it's a popular feature request: https://github.com/Budibase/budibase/issues/1372

**Reply by rafikApp** [reply]
And what about signature

**Reply by melohagan** [reply]
https://github.com/Budibase/budibase/issues?q=is%3Aissue+is%3Aopen+signature

**Reply by rafikApp** [reply]
Can I use javascript to create a form and save the fields to the database?

**Reply by melohagan** [reply]
No, there's no way to do that within Budibase, outside perhaps using an Embed.

**Reply by rafikApp** [reply]
Thank you @melohagan 
If I create a form with javascript using the embed component, how will I store the data entered by the user in the database?

**Reply by melohagan** [reply]
There's definitely no convenient way to do that. You would need to make an API request to the couchdb instance. I really don't recommend it though, because at that point it's not really low-code. You may as well contribute to the codebase the functionality you need, or build your own web-app from scratch with Svelte or React etc, if a lot of flexibility and custom functionality is needed.

**Reply by rafikApp** [reply]
if I work with a MySql type database, would I have more freedom or is it a limitation via javascript with the platform
